lib/vehicles_communicator.py

```
@@ -14,84 +16,95 @@ class VehiclesCommunicator:
-            logging.error("Failed to connect to Protocol HTTP API. Retrying.")
-            self._wait_till_api_is_available()
-            return self._send_request(url_postfix)
 
-    def get_position(self, car) -> Point | None:
-        request_url = f"/status/{car['company_name']}/{car['car_name']}"
-        car_status_json = self._send_request(request_url)
+    def _send_request(self, function):
+        try:
+            return function()
+        except UnauthorizedException:
+            raise self.AuthenticationException("Invalid API key.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)